ui/src/main/webapp/app/query/operation-chain-component.js

```diff
@@ -108,14 +107,22 @@ function OperationChainController(operationChain, config, loading, query, error,
             chain.operations.push(operationService.createDeduplicateOperation(options));
         }
 
-        
         previousQueries.addQuery({
             name: "Operation Chain",
             lastRun: moment().format('HH:mm'),
             operations: vm.operations
         });
 
         runQuery(chain.operations);
+
+        if (settings.getClearChainAfterExecution() == true) {
+            resetChainWithoutDialog();
```

Comment:
Having this code here, rather than in the callback in the runQuery function means that you can see the chain reset visibly. This slows down the execution of the query. I think it should appear to happen behind the scenes.

Reply:
Moved the clearing of the chain to after the data has loaded and changed the tests.